<?xml version="1.0"?>
<launch>
  <arg name="show_calibration_gui" default="false"/>
  <arg name="enable_table_camera" default="false"/>

  <!-- sensor placement (from calibration) -->
  <!-- Table-mounted camera -->
  <node name="robot_to_camera_publisher" pkg="tf" type="static_transform_publisher"
        args="0.7 0.0 0.1 0.5 0.0 -1.5708 iiwa_link_0 camera_color_optical_frame 100"
        if="$(arg enable_table_camera)"/>

  <!-- Wrist-mounted camera -->
  <node name="robot_to_camera_link" pkg="tf" type="static_transform_publisher"
        args="0.0 0.0 0.0 0.0 0.0 1.57079632679 camera_base_link camera_link 100"/>

  <node name="camera_to_cal_frame" pkg="tf" type="static_transform_publisher"
        args="0.0 0.0 0.0 0.0 3.1416 -1.57079632679 camera_color_frame camera_cal_frame 100"/>

  <node name="cal_to_camera_optical_frame" pkg="tf" type="static_transform_publisher"
        args="0.000 0.004 0.000 0.0 0.0 -0.033 camera_cal_frame camera_color_optical_frame 100"
        unless="$(arg show_calibration_gui)"/>

  <node name="manual_calibration_gui" type="tf_publisher_gui.py" pkg="vpr_tracking"
        if="$(arg show_calibration_gui)">
    <param name="rate" value="20.0"/>
    <param name="parent_frame" value="camera_cal_frame"/>
    <param name="child_frame" value="camera_color_optical_frame"/>
    <rosparam param="init_position">[0.0, 0.0, 0.0]</rosparam>
    <rosparam param="init_rotation">[0.0, 0.0, 0.0]</rosparam>
    <remap from="pose_st" to="cal_pose"/>
  </node>
</launch>
